variable/api.go

```diff
@@ -182,6 +182,9 @@ func setFlushedValue(ctx context.Context, index uint32, value interface{}) error
 			if setter == nil {
 				return errors.New(errSetterNotFound + variable.Name())
 			}
+
+			// should invalidate the cached value before setting it to a new one
+			variableValue.Valid = false
```

Comment:
What scenarios will lead to failure? I see `DefaultSetter` do not judge `Valid`.

Reply:
if we both have getter and setter, then the cached value will not be modified by setter and may return obsolete value